cmd/gateway/azure/gateway-azure.go

```diff
@@ -505,11 +512,109 @@ func (a *azureObjects) DeleteBucket(ctx context.Context, bucket string) error {
 	return azureToObjectError(err, bucket)
 }
 
+// Filter out the entries which are lexicographically <= marker and return the rest of the listing.
+func (a *azureObjects) filterAndListObjects(ctx context.Context, bucket, prefix, marker, delimiter string, maxKeys int) (result minio.ListObjectsInfo, err error) {
+	var objects []minio.ObjectInfo
+	var prefixes []string
+	container := a.client.GetContainerReference(bucket)
+	azureListMarker := ""
+	for {
+		resp, err := container.ListBlobs(storage.ListBlobsParameters{
+			Prefix:     prefix,
+			Marker:     azureListMarker,
+			Delimiter:  delimiter,
+			MaxResults: uint(maxKeys),
+		})
+		if err != nil {
+			logger.LogIf(ctx, err)
+			return result, azureToObjectError(err, bucket, prefix)
+		}
+		azureListMarker = resp.NextMarker
+		for i, object := range resp.Blobs {
+			if strings.HasPrefix(object.Name, minio.GatewayMinioSysTmp) {
+				continue
+			}
+			if object.Name <= marker {
+				continue
+			}
+			// All entries >= marker are returned to the client.
+			for _, object = range resp.Blobs[i:] {
+				if strings.HasPrefix(object.Name, minio.GatewayMinioSysTmp) {
+					continue
+				}
+				objects = append(objects, minio.ObjectInfo{
+					Bucket:          bucket,
+					Name:            object.Name,
+					ModTime:         time.Time(object.Properties.LastModified),
+					Size:            object.Properties.ContentLength,
+					ETag:            minio.ToS3ETag(object.Properties.Etag),
+					ContentType:     object.Properties.ContentType,
+					ContentEncoding: object.Properties.ContentEncoding,
+				})
+			}
+			break
+		}
+		for i, prefix := range resp.BlobPrefixes {
+			if prefix == minio.GatewayMinioSysTmp {
+				continue
+			}
+			if prefix <= marker {
+				continue
+			}
+			// All entries >= marker are returned to the client.
+			for _, prefix = range resp.BlobPrefixes[i:] {
+				if prefix == minio.GatewayMinioSysTmp {
+					continue
+				}
+				prefixes = append(prefixes, prefix)
+			}
+			break
+		}
+		if len(objects) > 0 || len(prefixes) > 0 {
+			break
+		}
+		if azureListMarker == "" {
```

Comment:
You could add a comment about the check something like `As we got empty next marker from Azure to denote no more entries available,  stop iterating again`